File: erlang/grains/grains.erl
-module(grains).
-export([square/1, total/0]).

square(Square) -> trunc(math:pow(2, Square-1)).

total() -> total(64, 0).
total(0, Acc) -> Acc;
% Recursion is not the best choice here
total(Square, Acc) -> ThisSquare = square(Square), total(Square-1, Acc + ThisSquare).

File: csharp/grains/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n < 1 || n > 64) {
            throw new ArgumentOutOfRangeException("nope");
        }
        return (ulong)Math.Pow(2, n-1);
    }

    public static ulong Total()
    {
        ulong result = 0;
        // Would have been nice to use .Sum but that's not defined for ulong
        foreach (ulong sq in Enumerable.Range(1, 64).Select(x => Square(x))) {
            result += sq;
        }

        return result;
    }
}